### PR TITLE
Correctly patch empty token

### PR DESCRIPTION
**Pull request by allanrenucci**

### CODE REVIEW

File: scalafix-core/src/main/scala/scalafix/syntax/package.scala [hunk 1]
```diff
@@ -71,6 +71,14 @@ package object syntax {
 
   implicit class XtensionToken(token: Token) {
     def posTuple: (Int, Int) = token.start -> token.end
+    // copy pasted from:
+    // github.com/scalameta/scalafmt/blob/master/scalafmt-core/shared/src/main/scala/org/scalafmt/util/TokenOps.scala
+    def hash: Long = {
```
Comment (allanrenucci) [inline scalafix-core/src/main/scala/scalafix/syntax/package.scala]: Maybe you don't want to expose this. I can inline it at the use site.

File: scalafix-core/src/main/scala/scalafix/syntax/package.scala [hunk 2]
```diff
@@ -71,6 +71,14 @@ package object syntax {
 
   implicit class XtensionToken(token: Token) {
     def posTuple: (Int, Int) = token.start -> token.end
```
Comment (allanrenucci) [inline scalafix-core/src/main/scala/scalafix/syntax/package.scala]: This is no longer used in the project

**Comment by allanrenucci** [comment]
Let me know if I put the tests at the right place and if the naming is OK.

File: scalafix-core/src/main/scala/scalafix/syntax/package.scala (hunk 2 above)
Comment (olafurpg) [inline scalafix-core/src/main/scala/scalafix/syntax/package.scala]: We can remove this.

File: scalafix-core/src/main/scala/scalafix/syntax/package.scala (hunk 1 above)
Comment (olafurpg) [inline scalafix-core/src/main/scala/scalafix/syntax/package.scala]: Can we make this `private [scalafix]`?

Which reminds me I should setup MiMa to keep track of the public api.

File: scalafix-tests/src/main/scala/scalafix/test/FqnRewrite.scala [hunk 3]
```diff
@@ -26,3 +26,12 @@ object LambdaRewrites {
   }
 
 }
+
+case object FqnRewrite3 extends Rewrite {
```
Comment (olafurpg) [inline scalafix-tests/src/main/scala/scalafix/test/FqnRewrite.scala]: Can we also give this rewrite a more descriptive name? For example, `PatchTokenWithEmptyRange`. The `FqnRewrite` names are for testing the classloading feature.

File: scalafix-core/src/main/scala/scalafix/syntax/package.scala [hunk 4]
```diff
@@ -70,7 +70,14 @@ package object syntax {
   }
 
   implicit class XtensionToken(token: Token) {
-    def posTuple: (Int, Int) = token.start -> token.end
+    // copy pasted from:
+    // github.com/scalameta/scalafmt/blob/master/scalafmt-core/shared/src/main/scala/org/scalafmt/util/TokenOps.scala
```
Comment (olafurpg) [inline scalafix-core/src/main/scala/scalafix/syntax/package.scala]: I just noticed this links to master branch, can you please canonicalize it?

Comment (allanrenucci) [inline scalafix-core/src/main/scala/scalafix/syntax/package.scala]: What do you mean?
scalameta/scalafmt/scalafmt-core/shared/src/main/scala/org/scalafmt/util/TokenOps.scala
Like this?

Comment (olafurpg) [inline scalafix-core/src/main/scala/scalafix/syntax/package.scala]: If you press "y" while browsing a file on github.com, the url will expand into something like this 

https://github.com/scalameta/scalafmt/blob/ba319e0aaee6d4e1b30098ad57940c67910c33e9/scalafmt-core/shared/src/main/scala/org/scalafmt/util/TokenOps.scala

this url will still work even if the file is edited/moved/removed

Comment (allanrenucci) [inline scalafix-core/src/main/scala/scalafix/syntax/package.scala]: Done

**Comment by olafurpg** [comment]
Thank you for fixing this!

**Pull request merged**